project/apps/api-gateway/src/collaboration/collaboration.controller.ts

```diff
@@ -8,5 +8,43 @@ export class CollaborationController {
     private readonly collaborationServiceClient: ClientProxy,
   ) {}
 
-  // TODO: Add route handling methods
+  @Get(':id')
+  async getCollaborationInfoById(@Param('id') collabId: string) {
+    return this.collaborationServiceClient.send(
+      { cmd: 'get_collab_info' },
+      collabId,
+    );
+  }
+
+  @Get('all/:id')
+  async getAllCollaborations(@Param('id') userId: string) {
+    return this.collaborationServiceClient.send(
+      { cmd: 'get_all_collabs' },
+      userId,
+    );
+  }
+
+  @Get('active/:id')
+  async getActiveCollaborations(@Param('id') userId: string) {
+    return this.collaborationServiceClient.send(
+      { cmd: 'get_active_collabs' },
+      userId,
+    );
+  }
```

Comment:
nit, seeking opinion: would it be better do use url params for this instead?
similar to how its done for questions

https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g30/blob/d443a6f4c28504cbf962e579f74081a73ed0b8d6/project/apps/question-service/src/adapters/controllers/questions.controller.ts#L15-L18

Reply:
Hmmm, not sure if i misunderstood your point but the section you highlighted is under api-gateway not the collab service directory, can you give me an example changing it similar to questions? Because from my end it appears to be handled similarly. I might have missed something out here

Reply:
oops sflr, yup i was referring to using the URL param filters instead of a `/all` route